internal/raterun/runner.go

```diff
@@ -1,102 +1,117 @@
 package raterun
 
 import (
+	"context"
 	"errors"
 	"time"
 )
 
-// RunFunction function to be executed at specified rate
-type RunFunction func(rate time.Duration, t time.Time)
+// RunFunction function to be executed every period
+type RunFunction func(period time.Duration)
 
-type Rate struct {
-	Start time.Duration // when to start this rate
-	Rate  time.Duration // run in amount of Duration
+type Schedule struct {
```

Comment:
nit: comments here explaining how the properties are used would help me.

// StartDelay is the delay that this schedule should start at.
// Frequency defines how often the provided RunFunction should be executed during this schedule.